byron/chain/executable-spec/src/Cardano/Spec/Chain/STS/Block.hs

```diff
@@ -94,18 +101,27 @@ bUpdPayload b = (b ^. bBody ^. bUpdProp, b ^. bBody ^. bUpdVotes)
 bEndorsment :: Block -> (ProtVer, VKey)
 bEndorsment b = (b ^. bBody ^. bProtVer, b ^. bHeader ^. bhIssuer)
 
-
 -- | Compute the size (in words) that a block takes.
 bSize :: Block -> Natural
-bSize = fromIntegral . abstractSize acctMap
+bSize b = bHeaderSize (b ^. bHeader) + bBodySize (b ^. bBody)
+
+-- | Compute the size (in words) that a block body occupies.
+bBodySize :: BlockBody -> Natural
+bBodySize = fromIntegral . abstractSize costs
   where
-    acctMap = []
+    costs = Map.fromList [ (typeOf (undefined::Maybe UProp), 1)
+                         , (typeOf (undefined::ProtVer), 1)
+                         , (typeOf (undefined::DCert), 1)
+                         , (typeOf (undefined::TxWits TxId), 1)]
```

Comment:
We're missing the costs for the `Vote`s inside the `Block`.

Reply:
And for the costs of the transactions I think we need to count:

- The individual `Wit`s
- The inputs
- The outputs

Reply:
I've added (Vote,1) and for the variable components in TxWits ->  Wit/TxIn/TxOut